test/flycheck-test.el

```diff
@@ -3518,6 +3518,27 @@ Why not:
    '(6 6 error "Glob written as <...> (See page 167 of PBP)"
        :id "BuiltinFunctions::RequireGlobFunction" :checker perl-perlcritic)))
 
+(flycheck-ert-def-checker-test perl perl modules
+  ;; Files that require unlisted modules should fail to check
+  (flycheck-ert-should-syntax-check
+   "language/perl/Script.pl" '(perl-mode cperl-mode)
+   '(3 nil error "Global symbol \"$dependency_a\" requires explicit package name (did you forget to declare \"my $dependency_a\"?)"
+       :checker perl)
+   '(4 nil error "Global symbol \"$dependency_b\" requires explicit package name (did you forget to declare \"my $dependency_b\"?)"
+       :checker perl))
+  ;; Including those modules should allow them to check
+  (cl-letf
```

Comment:
It did not occur to me we could call `-should-syntax-check` multiple times in the same test.  Is it just to regroup the checks under the same 'module' test?  Or are there any other benefits?  (Just curious; it's fine as is)

Why `cl-letf` though?  Wouldn't a simple `let` would suffice to override the variable for the next call to `-should-syntax-check`?